File: example/BBoxOne/first_generation/C#/ConsoleApp1/BBOXONE_DEMO.cs
using System;
using System.Linq;
using System.Collections.Generic;
using BBoxAPI;


namespace BBOXONE_DEMO
{
    class BBOXONE_CONTROL_DEMO
    {
        const int scanning_mode = 0;

        const int TX = 0;
        const int RX = 1;

        static double[,] DR = null;

        static double TX_MIN_GAIN = 0;
        static double TX_MAX_GAIN = 0;
        static double RX_MIN_GAIN = 0;
        static double RX_MAX_GAIN = 0;

        static string[] AAkitList = null;


        static void Main(string[] args)
        {
            string sn = null;
            string[] dev_info = null;
            int DEV_NUM = 0;

            BBoxOneAPI instance = new BBoxOneAPI();

            dev_info = instance.ScanningDevice(scanning_mode);
            
            DEV_NUM = dev_info.Count();

            for (int i = 0; i < DEV_NUM; i++)
            {
                string[] response_message = dev_info[i].Split(',');
                sn = response_message[0];
                instance.Init(sn, 0, i);

                string[] freq_list = instance.getFrequencyList(sn);

                instance.setOperatingFreq(Convert.ToDouble(freq_list[0]), sn);

                DR = instance.getDR(sn);
                TX_MIN_GAIN = DR[0, 0];
		        TX_MAX_GAIN = DR[0, 1];
		        RX_MIN_GAIN = DR[1, 0];
		        RX_MAX_GAIN = DR[1, 1];

                AAkitList = instance.getAAKitList(sn);
		        instance.selectAAKit(AAkitList[0], sn);

                Console.WriteLine("======================================================");
		        Console.WriteLine("[DEMO1] Switch TX mode");
		        Console.ReadKey();
                instance.SwitchTxRxMode(TX, sn);
                int mode = instance.getTxRxMode(sn);
		        Console.WriteLine("[DEMO1] Mode : " + mode);

                Console.WriteLine("======================================================");
		        Console.WriteLine("[DEMO2] Channel power control : Off channel 1 power");
		        Console.ReadKey();

                int board = 1;
		        int channel = 1;
		        int sw = 1;
		        instance.switchChannelPower(board, channel, sw, sn);
		        Console.WriteLine("[DEMO2] Channel 1 power off");

                Console.WriteLine("======================================================");
		        Console.WriteLine("[DEMO3] Channel Gain/Phase Control");
		        Console.WriteLine("[DEMO3] Channel_1 gain  : 15 db");
		        Console.WriteLine("[DEMO3] Channel_1 phase : 15 deg");
		        Console.WriteLine("[DEMO3] Channel_2 gain  : 15 db");
		        Console.WriteLine("[DEMO3] Channel_2 phase : 30 deg");
		        Console.WriteLine("[DEMO3] Channel_3 gain  : 15 db");
		        Console.WriteLine("[DEMO3] Channel_3 phase : 45 deg");
		        Console.WriteLine("[DEMO3] Channel_4 gain  : 15 db");
		        Console.WriteLine("[DEMO3] Channel_4 phase : 60 deg");
		        Console.ReadKey();

                board = 1;

		        channel = 1;
		        double Target_db = 15;
		        int Target_deg = 15;
		        instance.setChannelGainPhase(board, channel, Target_db, Target_deg, sn);

		        channel = 2;
		        Target_db = 15;
		        Target_deg = 30;
		        instance.setChannelGainPhase(board, channel, Target_db, Target_deg, sn);

		        channel = 3;
		        Target_db = 15;
		        Target_deg = 45;
		        instance.setChannelGainPhase(board, channel, Target_db, Target_deg, sn);

		        channel = 4;
		        Target_db = 15;
		        Target_deg = 60;
		        instance.setChannelGainPhase(board, channel, Target_db, Target_deg, sn);

                Console.WriteLine("======================================================");
		        Console.WriteLine("[DEMO4] BeamSteering Control");
		        Console.ReadKey();
		        Console.WriteLine("[DEMO4] Channel gain  : 12 db");
		        Console.WriteLine("[DEMO4] Beam angle : 15 ");

		        Target_db = 12;
		        int Target_angle_x = 15;
		        int Target_angle_y = 0;
		        instance.setBeamXY(Target_db, Target_angle_x, Target_angle_y, sn);
            }

            Console.WriteLine("======================================================");
	        Console.WriteLine("[DEMO] End");
		    Console.ReadKey();
        }
    }
}

File: example/BBoxOne Series/BBoxOne  5G/Python/BBoxOne_DEMO.py
import sys
import clr
import time
import csv
import os
import numpy as np

dir_path = '..\\..\\..\\..\\'
os.chdir(os.path.abspath(dir_path))

path = '.\\BBoxAPI.dll'
clr.AddReference(os.path.abspath(path))

from BBoxAPI import *

if __name__ == '__main__':

	TX = 1
	RX = 2

	AAkitList = []

	instance = BBoxOneAPI()

	dev_info = instance.ScanningDevice(0)
	device_num = len(dev_info)

	if device_num > 0:
		for i in range(0, device_num, 1):

			response_message = dev_info[i].split(",")
			sn = response_message[0]
			ip = response_message[1]
			val = response_message[2].split("\x00")
			dev_type = int(val[0])

			print("[BBoxOne_DEMO][GetDeviceStatus] SN : %s" % (sn))
			print("[BBoxOne_DEMO][GetDeviceStatus] IP : %s" % (ip))
			print("[BBoxOne_DEMO][GetDeviceStatus] dev_type : %d" % (dev_type))

			instance.Init(sn, dev_type, i)
			print("[BBoxOne_DEMO][InitialDevice][%s]" % (sn))

			freq_list = instance.getFrequencyList(sn)
			if len(freq_list) == 0:
				print("[BBoxOne_DEMO][InitialDevice][%s] failed : return null" % (sn))
			else:
				for item in freq_list:
					print("[BBoxOne_DEMO][InitialDevice][%s] getFrequencyList: list %s" % (sn, item))

			ret = instance.setOperatingFreq(freq_list[0], sn)
			if ret != 0:
				print("[BBoxOne_DEMO][InitialDevice][%s] setOperatingFreq failed : error_code %d" % (sn, ret))

			DR = instance.getDR(sn)
			print("[BBoxOne_DEMO][InitialDevice][%s] " % (sn))
			TX_MIN_GAIN = DR[0, 0]
			print("[BBoxOne_DEMO][InitialDevice][%s] TX_MIN_GAIN : %f" % (sn, TX_MIN_GAIN))
			TX_MAX_GAIN = DR[0, 1]
			print("[BBoxOne_DEMO][InitialDevice][%s] TX_MAX_GAIN : %f" % (sn, TX_MAX_GAIN))
			RX_MIN_GAIN = DR[1, 0]
			print("[BBoxOne_DEMO][InitialDevice][%s] RX_MIN_GAIN : %f" % (sn, RX_MIN_GAIN))
			RX_MAX_GAIN = DR[1, 1]
			print("[BBoxOne_DEMO][InitialDevice][%s] RX_MAX_GAIN : %f" % (sn, RX_MAX_GAIN))

			AAkitList.append(instance.getAAKitList(sn))
			instance.selectAAKit(AAkitList[0][0], sn)

			print("======================================================")
			print("[BBoxOne_DEMO][DEMO1][%s] Switch TX mode" % (sn))
			os.system("pause")
			instance.SwitchTxRxMode(TX, sn)
			mode = instance.getTxRxMode(sn)
			print("[BBoxOne_DEMO][DEMO1][%s] Mode : %d" % (sn, mode))

			print("======================================================")
			print("[BBoxOne_DEMO][DEMO2][%s] Channel power control : Off channel 1 power" % (sn))
			os.system("pause")

			board = 1
			channel = 1
			sw = 1
			instance.switchChannelPower(board, channel, sw, sn)
			print("[BBoxOne_DEMO][DEMO2][%s] Channel 1 power off" % (sn))

			Target_db = TX_MAX_GAIN
			Target_ch1_deg = 15
			Target_ch2_deg = 30
			Target_ch3_deg = 45
			Target_ch4_deg = 60
   
			print("======================================================")
			print("[BBoxOne_DEMO][DEMO3][%s] Channel Gain/Phase Control" % (sn))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_1 gain  : %f db" % (sn, Target_db))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_1 phase : %d deg" % (sn, Target_ch1_deg))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_2 gain  : %f db" % (sn, Target_db))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_2 phase : %d deg" % (sn, Target_ch2_deg))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_3 gain  : %f db" % (sn, Target_db))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_3 phase : %d deg" % (sn, Target_ch3_deg))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_4 gain  : %f db" % (sn, Target_db))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_4 phase : %d deg" % (sn, Target_ch4_deg))
			os.system("pause")

			board = 1

			channel = 1
			instance.setChannelGainPhase(board, channel, Target_db, Target_ch1_deg, sn)

			channel = 2
			instance.setChannelGainPhase(board, channel, Target_db, Target_ch2_deg, sn)

			channel = 3
			instance.setChannelGainPhase(board, channel, Target_db, Target_ch3_deg, sn)

			channel = 4
			instance.setChannelGainPhase(board, channel, Target_db, Target_ch4_deg, sn)

			Target_db = TX_MAX_GAIN - 1
			Target_ch1_deg = 45
			Target_ch2_deg = 60
			Target_ch3_deg = 75
			Target_ch4_deg = 90

			print("[BBoxOne_DEMO][DEMO3][%s] Channel Gain/Phase Control" % (sn))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_5 gain  : %f db" % (sn, Target_db))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_5 phase : %d deg" % (sn, Target_ch1_deg))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_6 gain  : %f db" % (sn, Target_db))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_6 phase : %d deg" % (sn, Target_ch2_deg))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_7 gain  : %f db" % (sn, Target_db))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_7 phase : %d deg" % (sn, Target_ch3_deg))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_8 gain  : %f db" % (sn, Target_db))
			print("[BBoxOne_DEMO][DEMO3][%s] Channel_8 phase : %d deg" % (sn, Target_ch4_deg))
			os.system("pause")

			board = 2

			channel = 1
			instance.setChannelGainPhase(board, channel, Target_db, Target_ch1_deg, sn)

			channel = 2
			instance.setChannelGainPhase(board, channel, Target_db, Target_ch2_deg, sn)

			channel = 3
			instance.setChannelGainPhase(board, channel, Target_db, Target_ch3_deg, sn)

			channel = 4
			instance.setChannelGainPhase(board, channel, Target_db, Target_ch4_deg, sn)

			print("======================================================")
			print("[BBoxOne_DEMO][DEMO4][%s] BeamSteering Control" % (sn))
			os.system("pause")
   
			Target_db = TX_MAX_GAIN - 2
			Target_theta = 15
			Target_phi = 0
   
			print("[BBoxOne_DEMO][DEMO4][%s] Channel gain  : %f db" % (sn, Target_db))
			print("[BBoxOne_DEMO][DEMO4][%s] Theta : %d " % (sn, Target_theta))
			print("[BBoxOne_DEMO][DEMO4][%s] Phi : %d " % (sn, Target_phi))

			instance.setBeamAngle(Target_db, Target_theta, Target_phi, sn)

	print("======================================================")
	print("[BBoxOne_DEMO][DEMO] End")
	os.system("pause")
